### ISSUE TITLE
Shebang support

### ISSUE THREAD

**Issue by brandonkal** [DavidAnson/markdownlint#310]
The file's shebang line should be stripped before the text is passed to rules.
I found that rules 18 and 41 complain on the line. This should be simple to implement.

**Comment by DavidAnson** [comment]
Can you please show an example? I’m not familiar with shebang in Markdown; the files are not usually executable.

https://en.m.wikipedia.org/wiki/Shebang_(Unix)

**Comment by brandonkal** [comment]
The pattern is not widespread but it enables some exciting use-cases.
- I support it in my https://github.com/brandonkal/inkjet tool to build rapid CLIs and as a project task runner.
- There is also https://github.com/bashup/mdsh to compile markdown to bash scripts.

**Comment by DavidAnson** [comment]
It’s a slight abuse of the feature, but have you tried defining a front matter regular expression to match the shebang syntax? That should tell the tool to strip that first line out automatically.

**Comment by DavidAnson** [comment]
For this use case, I the suggestion above seems to work fine:

```bash
pi@pi:~/shebang $ cat shebang.md
#!/usr/bin/env node

# Heading

Markdown text.
pi@pi:~/shebang $ cat .markdownlint-cli2.jsonc
{
  "frontMatter": "^#!.*(\r?\n)"
}
pi@pi:~/shebang $ markdownlint-cli2 shebang.md
Finding: shebang.md
Linting: 1 file(s)
Summary: 0 error(s)
pi@pi:~/shebang $
```

**Comment by DavidAnson** [comment]
Closing issue since this seems to be an uncommon scenario and the approach outlined above works.
